src/js/DatePicker.js

```diff
@@ -0,0 +1,436 @@
+            if ($target.is('a.ui-state-default')) {
+                keyEvent.preventDefault();
+
+                const $firstOfMonth = $target.closest('tbody').find('.ui-state-default').first();
+                if ($firstOfMonth) {
+                    $firstOfMonth.trigger('focus');
+                    this.highlightUtils.setHighlightState($firstOfMonth, $container);
+                }
+            }
+        } else if (keyCode === keyCodes.END) {
+            if ($target.is('a.ui-state-default')) {
+                keyEvent.preventDefault();
+
+                const $lastDay = $target.closest('tbody').find('.ui-state-default').last();
+                if ($lastDay) {
+                    $lastDay.trigger('focus');
+                    this.highlightUtils.setHighlightState($lastDay, $container);
+                }
+            }
+        }
+    }
+
+    /**
+     * Handle calendar close
+     */
+    closeCalendar () {
+        // Remove bound handler
+        this.$html.find('#ui-datepicker-div').off('keydown');
+        // Close the date picker
+        this.$dateInput.datepicker('hide');
+        // Return focus to the date input
+        this.$dateInput.trigger('focus');
+    }
+
+    /**
+     * Handle next month link click event
+     * @param {Event} event
+     */
+    handleNextLinkClicks (event) {
+        if (event !== undefined) {
+            event.preventDefault();
+        }
+
+        // Update the header elements after change of month
+        this.updateHeaderElements();
+        // Highlight date in new month
+        this.highlightUtils.prepHighlightState(this.$datePickerContainer);
+        // Refocus the next button (otherwise focus returns to date input)
+        this.$html.find('.ui-datepicker-next').trigger('focus');
+        // Re-hide the today button
+        this.$html.find('.ui-datepicker-current').addClass('u-display-none');
+    }
+
+    /**
+     * Handle previous month link click event
+     * @param {Event} event
+     */
+    handlePrevLinkClicks (event) {
+        if (event !== undefined) {
+            event.preventDefault();
+        }
+
+        // Update the header elements after change of month
+        this.updateHeaderElements();
+        // Highlight date in new month
+        this.highlightUtils.prepHighlightState(this.$datePickerContainer);
+        // Refocus the prev button (otherwise focus returns to date input)
+        this.$html.find('.ui-datepicker-prev').trigger('focus');
+        // Re-hide the today button
+        this.$html.find('.ui-datepicker-current').addClass('u-display-none');
+    }
+
+    /**
+     * Update calendar header elements after month change
+     */
+    updateHeaderElements () {
+        // Re-get the next/prev buttons as they change when the month changes
+        this.$prevLink = this.$datePickerContainer.find('.ui-datepicker-prev');
+        this.$nextLink = this.$datePickerContainer.find('.ui-datepicker-next');
+
+        // Make them focusable
```

Comment:
there's some duplication here and in the `enhanceDatePicker` method, perhaps one to be DRYed out?

Reply:
You could achieve this by adding the links into an array and looping over them to do all of the changes required